src/porepy/models/abstract_equations.py

```diff
@@ -1,18 +1,78 @@
-"""Abstract equation classes.
+"""Module containing classes representing generalized equations and variables.
+
+- :class:`EquationMixin`: Base equation class providing the interface for setting
+    equations. This interface is used by the solution strategy mixin.
+- :class:`BalanceEquation`: Intermediate base class for vector and scalar balance
+    equations.
+- :class:`LocalElimination`: Intermediate base class for local equations which
+    represent the elimination of some variable by a correlation/function dependent on
+    other variables.
+- :class:`VariableMixin`: Base class for variables, providing the interface for
+    creating variables. This interface is used by the solution strategy mixin.
 
-Contains:
-    - BalanceEquation: Base class for vector and scalar balance equations.
-    - VariableMixin: Base class for variables.
 """
 
 from __future__ import annotations
 
-from typing import Union, cast
+from functools import cached_property
+from typing import Callable, Sequence, Union, cast
+
+import numpy as np
 
 import porepy as pp
+from porepy.numerics.ad.equation_system import GridEntity
+
+
+class EquationMixin(pp.PorePyModel):
+    """General class for equations defining an interface to introduce equations into
+    a model.
+
+    Example:
+        When creating multi-physics models, the order of equations will reflect
+        the order of inheritance of individual equation classes.
+
+        Say we have to equation classes combined into one multi-physics model:
+
+        .. code::python
+
+            class Equation1(EquationMixin):
+
+                def set_equations(self):
+                    super().set_equation()
+                    # proceed to set equation 1
+
+            class Equation2(EquationMixin):
+
+                def set_equations(self):
+                    super().set_equation()
+                    # proceed to set equation 2
+
+            class ModelEquations(Equation1, Equation2):
+                pass
+
+        Notice that in both ``set_equations`` the super-call comes first.
+        The way ``super()`` works now is, that ``Equation1`` will be called first,
+        which in return executes the code of ``Equation2`` first.
+        I.e., the order of equations introduced into the system is the reverse order
+        of equation classes in the inheritance tree of the combined class.
+
+        Notice also, that the combined class ``ModelEquations`` does not need an
+        implementation of ``set_equations``.
+
+    """
+
+    def set_equations(self) -> None:
+        """Method to be overridden to set equations on some grid.
+
+        The base class method does nothing and is implemented to provide the right
+        signature and to help the super call to resolve the setting of equations in
+        multi-physics models.
+
+        """
+        pass
 
 
-class BalanceEquation(pp.PorePyModel):
+class BalanceEquation(EquationMixin):
     """Generic class for vector balance equations.
```

Comment:
Any idea why this says "vector"? It seems something has gone wrong with the documentation in a (previous) refactoring.

Reply:
My guess would be because of the mechanics models, which also inherit this one. Also, the volume integral allows users to define the mathematical dimension of the integrated quantity.

Reply:
Yes, but wouldn't "scalar or vector" be clearer?

Reply:
I'll change.